woocommerce/woocommerce-android
# Purchase shipping label

Closes: #12287

### Description
This PR adds support for purchasing a shipping label. The changes introduced here include network support, data transformation, blocking the UI while the purchase request is in progress, and navigation to the final screen.

### Testing information

1. Open the app
2. Navigate to the Orders section
3. Open an order eligible for shipping label creation
4. Tap on create shipping label
5. Tap on select a package
6. Select any package
7. Tap on a shipping rate
8. Check that the purchase label button is enabled
9. Tap on the purchase label
10. Check that the UI is blocked while the purchase is in progress
11. Check that the app navigates to the final screen when the purchase completes

### The tests that have been performed
1. Checked that the request is made to the API
2. Checked that the app navigates to the final screen after the request completes

### Images/gif

https://github.com/user-attachments/assets/3aab40e4-989f-4189-b59e-2aa25fb99f56


## Reviewer (or Author, in the case of optional code reviews):

Please make sure these conditions are met before approving the PR, or request changes if the PR needs improvement:
